pkg/controller/utils/utils.go

```diff
@@ -986,3 +986,53 @@ func RemoveInstallationFinalizer(i *operatorv1.Installation, finalizer string) {
 		i.SetFinalizers(stringsutil.RemoveStringInSlice(finalizer, i.GetFinalizers()))
 	}
 }
+
+// MaintainInstallationFinalizer manages a controller's finalizer on the Installation resource.
+// We add a finalizer to the Installation when the mainResource has been installed, and only remove that finalizer when
+// the resource has been deleted and its secondary resources have stopped running. This allows for a graceful cleanup of any resources
+// prior to the CNI plugin being removed.
+func MaintainInstallationFinalizer(
```

Comment:
We should update the apiserver and installation core controllers to use this helper function instead of the duplicate ones they have there.